build-aux/Jenkinsfile.full

```diff
@@ -156,17 +157,12 @@ pipeline {
         stage('CentOS 6') {
           agent {
             docker {
-              image 'couchdbdev/centos-6-erlang-19.3.6:latest'
+              image 'couchdbdev/centos-6-erlang-20.3.8.24-1:latest'
+              args "${DOCKER_ARGS}"
               alwaysPull true
-              label 'ubuntu'
-              // this keeps builds landing on the same host from clashing with each other
-              customWorkspace pwd() + '/centos6'
+              customWorkspace './centos6'
```

Comment:
Is dropping the `pwd()` a stylistic change or are you actually moving the workspace to a different location than we had previously? [JENKINS-17818](https://issues.jenkins-ci.org/browse/JENKINS-17818) made me think that the `customWorkspace` is defined relative to `JENKINS_HOME`

Reply:
The problem is that this single `pwd()` call actually requires a top-level agent, meaning that for a 3-wide pipeline, that actually instantiates a fourth agent *just* to calculate `pwd()`, and does not terminate for the length of the run.

I know a while back I was worried about network traffic syncing changes between agents, but I've not seen the issue in a while. As long as each build is happening in its own subdirectory, I think we're fine, and that `pwd()` is probably only returning `/home/jenkins/path/to/workspace` anyway.

Reply:
@kocolosk if you want to check exactly where it's putting those builds, those IBM agents @davisp set up should have the remnants of a few builds on this PR now. With that perhaps we can clear up the problem.

Reply:
I'm not sure @davisp added any SSH pub keys on those instances besides his own (at least, I don't know how to log in), but yes let's check. I'm worried that the current configuration might be skipping the `/path/to/workspace` piece.

Reply:
For example, if I check the logs of a currently running job:

https://jenkins-cm1.apache.org/job/jenkins-cm1/job/FullPlatformMatrix/job/jenkins-pr-jenkinsfile/13/consoleFull

I see a line for CentOS 6 that says

> `16:35:57  Running on couchdb-ci-worker-dal-1-5 in /home/****/workspace/ormMatrix_****-pr-****file@3`

and then a few lines later one that seems to report the invocation of the `customWorkspace` command:

> `16:35:57  Running in /home/****/centos6@2`

Comparing these two suggests to me that we've exited the workspace entirely with that `customWorkspace` argument.

Reply:
OK, suggest an alternative. `pwd()` sacrificing an entire agent for the lifetime of the run seems a huge waste of resources.

Reply:
@kocolosk The list of SSH keys installed is here:

https://github.com/apache/couchdb-infra-cm/blob/main/roles/common/tasks/main.yml#L6-L8

I can add yours if you like.

Reply:
OK, I will see what I can do. I don't have my dev environment set up to test it right now, but I'm wondering about whether `${WORKSPACE}` can be used in the `customWorkspace` argument in lieu of `pwd()`

Reply:
No good.

```
groovy.lang.MissingPropertyException: No such property: WORKSPACE for class: groovy.lang.Binding
```

Reply:
Bummer. Thanks for checking. Will think on it a bit more.

Reply:
worse, https://issues.jenkins-ci.org/browse/JENKINS-60557

Reply:
An alternative is something like

```
customWorkspace "/path/to/well/known/OK/location/prbuild/${BRANCH_NAME}/centos6"
```

Reply:
Tried and didn't work:
* `${env.WORKSPACE}` (resolves to `null`)
* "${JENKINS_HOME}/workspace/${ITEM_FULL_NAME}" (see https://github.com/jenkinsci/jenkins/pull/3364/files#diff-9e24c3bccc9e921f1a5cc216abc25d03R5111 and https://issues.jenkins-ci.org/browse/JENKINS-21942) - "No such property: ITEM_FULL_NAME"
* "${JENKINS_HOME}/workspace/${ITEM_FULLNAME}" "groovy.lang.MissingPropertyException: No such property: ITEM_FULLNAME" (same as above)

@kocolosk According to Andrew Bayer, ["The problem here is that we generate the custom workspace string before we're actually on the agent."](https://issues.jenkins-ci.org/browse/JENKINS-50907?focusedCommentId=353326&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-353326) And since you can have custom paths per machine for where the workspace is stored, I think this may be intractable. Our approach previously was probably working because the scripted pipeline `pwd()` command was spinning up an agent, calculating that path, and it just happened to work everywhere (since every agent had the exact same file system layout). And, prior to your change, my `/tmp/...` was also consistent across all machines.

As a reminder, what we're working around here is https://issues.jenkins-ci.org/browse/JENKINS-57454

However, we may be chasing our tails, see https://issues.jenkins-ci.org/browse/JENKINS-48319

What about changing the build steps to `cd ${ERLANG_VERSION}` or `cd ${PLATFORM}` before they continue and give up on customWorkspaces entirely?

Reply:
For what it's worth, that last suggestion seems to work for now.

https://ci-couchdb.apache.org/blue/organizations/jenkins/jenkins-cm1%2FPullRequests/detail/PR-2367/37/pipeline

If you're OK with that @kocolosk I'll resolve this. Then we're just waiting on Infra who probably are still out until next week (as, technically, I am...)